Add SELinux variant to {busybox,f2fs-tools}

aparcar opened the pull request:
This patch accompanies the main [SELinux support](https://github.com/openwrt/openwrt/pull/3207) pull request by adding SELinux variants to both busybox and procd.

The move of Busyboxs config might be be a bit controversial.

aparcar committed:
busybox: add selinux variant

This commit adds a `selinux` variant which comes with with a number of
SELinux applets and also SELinux label support.

Signed-off-by: Paul Spooren <mail@aparcar.org>

dangowrt commented on package/utils/busybox/selinux.config:
```diff
@@ -0,0 +1,16 @@
+CONFIG_SELINUX=y
+CONFIG_FEATURE_TAR_SELINUX=y
+CONFIG_FEATURE_TAR_SELINUX=y
```
No need to have that line twice I guess

dangowrt commented on package/utils/busybox/Makefile:
```diff
@@ -9,15 +9,15 @@ include $(TOPDIR)/rules.mk
 PKG_HASH:=d0f940a72f648943c1f2211e0e3117387c31d765137d92bd8284a3fb9752a998
 
-PKG_BUILD_DEPENDS:=BUSYBOX_CONFIG_PAM:libpam
+PKG_BUILD_DEPENDS:=BUSYBOX_CONFIG_PAM:libpam BUSYBOX_CONFIG_SELINUX:libselinux
```
I don't think we need to change that line when using variants. It's most likely a left-over from the previous approach just adding the config symbols.

aparcar committed:
f2fs-tools: add selinux variant

This variant is build with `libselinux` and required to set labels
during runtime.

Signed-off-by: Paul Spooren <mail@aparcar.org>

dangowrt commented on package/utils/busybox/Makefile:
```diff
@@ -39,14 +39,27 @@ BUSYBOX_IF_ENABLED=$(if $(CONFIG_BUSYBOX_$(BUSYBOX_SYM)_$(1)),$(2))
   URL:=http://busybox.net/
   DEPENDS:=+BUSYBOX_CONFIG_PAM:libpam +BUSYBOX_CONFIG_NTPD:jsonfilter
-  MENU:=1
+  PROVIDES:=busybox
```
Move the `PROVIDES:=busybox` to busybox-selinux please, as busybox itself doesn't need that.

aparcar replied:
Done

dangowrt commented:
Applied, thanks!
